avro2tf/src/test/scala/com/linkedin/avro2tf/jobs/Avro2TFTest.scala

```diff
@@ -137,4 +140,84 @@ class Avro2TFTest extends WithLocalSparkSession {
+      close()
+    }
+
+    val avro2TFTrainingParams = Avro2TFJobParamsParser.parse(TestUtil.convertParamMapToParamList(
+      Map(
+        Avro2TFJobParamNames.INPUT_PATHS -> INPUT_TEXT_FILE_PATHS,
+        Avro2TFJobParamNames.WORKING_DIR -> workingDirectory,
+        Avro2TFJobParamNames.AVRO2TF_CONFIG_PATH -> avro2TFConfig,
+        Avro2TFJobParamNames.EXTERNAL_FEATURE_LIST_PATH -> externalFeatureListFullPath
+      )
+    ))
+
+    val workingDirectoryFilterZero = "/tmp/avro2tf-test-filterZero"
+    FileUtils.deleteDirectory(new File(workingDirectoryFilterZero))
+
+    val avro2TFTrainingParamsEnableFilterZero = Avro2TFJobParamsParser.parse(TestUtil.convertParamMapToParamList(
+      Map(
+        Avro2TFJobParamNames.INPUT_PATHS -> INPUT_TEXT_FILE_PATHS,
+        Avro2TFJobParamNames.WORKING_DIR -> workingDirectoryFilterZero,
+        Avro2TFJobParamNames.AVRO2TF_CONFIG_PATH -> avro2TFConfig,
+        Avro2TFJobParamNames.EXTERNAL_FEATURE_LIST_PATH -> externalFeatureListFullPath,
+        Avro2TFJobParamNames.ENABLE_FILTER_ZERO -> "true"
+      )
+    ))
+
+
+    Avro2TF.run(session, avro2TFTrainingParams)
+    assertTrue(new File(s"${avro2TFTrainingParams.workingDir.trainingDataPath}/_SUCCESS").exists())
+    assertTrue(new File(avro2TFTrainingParams.workingDir.tensorMetadataPath).exists())
+    assertTrue(new File(avro2TFTrainingParams.workingDir.featureListPath).exists())
+
+    val output = IOUtils.readAvro(session, getAvroFiles(avro2TFTrainingParams.workingDir.trainingDataPath): _*)
+    val last = output.collect().last
+
+    Avro2TF.run(session, avro2TFTrainingParamsEnableFilterZero)
+    assertTrue(new File(s"${avro2TFTrainingParamsEnableFilterZero.workingDir.trainingDataPath}/_SUCCESS").exists())
+    assertTrue(new File(avro2TFTrainingParamsEnableFilterZero.workingDir.tensorMetadataPath).exists())
+    assertTrue(new File(avro2TFTrainingParamsEnableFilterZero.workingDir.featureListPath).exists())
+
+    val outputFilterZero = IOUtils.readAvro(session, getAvroFiles(avro2TFTrainingParamsEnableFilterZero.workingDir.trainingDataPath): _*)
+    val lastFilterZero = outputFilterZero.collect().last
+
+    assertEquals(last.schema, lastFilterZero.schema)
+
+    val sparseVectorIndices = last.schema.filter(elem => CommonUtils.isSparseVector(elem.dataType)).map(last.schema.indexOf(_))
+
+    sparseVectorIndices.foreach {
+      index =>
+        assertFalse(last.get(index).asInstanceOf[GenericRowWithSchema].get(1).asInstanceOf[mutable.WrappedArray[Float]].filter(_ == 0.0).isEmpty)
+        assertTrue(lastFilterZero.get(index).asInstanceOf[GenericRowWithSchema].get(1).asInstanceOf[mutable.WrappedArray[Float]].filter(_ == 0.0).isEmpty)
+    }
+  }
+
+  def getAvroFiles(directory: String): List[String] = {
+    val d = new File(directory)
+    if (d.exists && d.isDirectory) {
+      d.listFiles.filter(_.isFile).map(_.getAbsolutePath).filter(_.endsWith("avro")).toList
+    } else {
+      List()
+    }
+  }
```

Comment:
not required, spark can directly read a folder

Reply:
Fixed in latest commit